### PR TITLE
Fix freeze on Linux

### PR DESCRIPTION
**Pull request by nightblade9**

### CODE REVIEW

**Comment by AbrarSyed** [comment]
Please note that I have take the liberty to rebase your branch over master and force-pushed it back up. In order to pull this branch please run `git reset --hard HEAD~3 && git pull`

File: AutoDungeoners.Web/Services/AbstractService.cs [hunk 1]
```diff
@@ -10,9 +10,10 @@ namespace AutoDungeoners.Web.Services
     /// a method to implement service-specific logic based on the amount of time elapsed.
     /// </summary>
     // Mostly copied from https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-3.1&tabs=netcore-cli#timed-background-tasks
-    abstract class AbstractService : IAbstractService, IHostedService
+    abstract class AbstractService : IAbstractService, IHostedService, IDisposable
```
Comment (AbrarSyed) [inline AutoDungeoners.Web/Services/AbstractService.cs:13]: is this really still an `AbstractService` if it has implementation specific functionality like a timer in it?

Comment (nightblade9) [reply AutoDungeoners.Web/Services/AbstractService.cs:13]: Yes - it's for abstract services that happen every tick. This is from back when we thought of using separate services for separate operations, but agreed that race conditions etc. make it simpler to build monolithic first and split out later. 

I can merge it into the main service if you like.

**Comment by AbrarSyed** [comment]
Nah its fine, was just asking

On May 21, 2020 1:40:29 AM UTC, nightblade9 <notifications@github.com> wrote:
>@nightblade9 commented on this pull request.
>
>
>
>> @@ -10,9 +10,10 @@ namespace AutoDungeoners.Web.Services
>/// a method to implement service-specific logic based on the amount of
>time elapsed.
>     /// </summary>
>// Mostly copied from
>https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?view=aspnetcore-3.1&tabs=netcore-cli#timed-background-tasks
>-    abstract class AbstractService : IAbstractService, IHostedService
>+    abstract class AbstractService : IAbstractService, IHostedService,
>IDisposable
>
>Yes - it's for abstract services that happen every tick. This is from
>back when we thought of using separate services for separate
>operations, but agreed that race conditions etc. make it simpler to
>build monolithic first and split out later. 
>
>I can merge it into the main service if you like.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/nightblade9/autodungeoners/pull/6#discussion_r428399400

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Pull request merged**